doc/modules/neural_networks_supervised.rst

```diff
@@ -0,0 +1,357 @@
+.. _neural_network:
+
+==================================
+Neural network models (supervised)
+==================================
+
+.. currentmodule:: sklearn.neural_network
+
+
+.. _multilayer_perceptron:
+ 
+Multi-layer Perceptron
+======================
+
+**Multi-layer Perceptron (MLP)** is a supervised learning algorithm that learns
+a function :math:`f(\cdot): R \rightarrow R` by training on a dataset. Given a 
+set of features :math:`X = {x_1, x_2, ..., x_m}` and a target :math:`y`, it 
+can learn a non-linear, complex functions for either classification
+or regression. It is different from logistic regression, in that between the input 
+and the output layer, there can be one or more hidden layers. Figure 1 shows 
+a one hidden layer MLP.
+
+.. figure:: ../images/multilayerperceptron_network.png
+   :align: center
+   :scale: 60%
+
+   **Figure 1 : One hidden layer MLP.**
+
+The leftmost layer, known as the input layer, consists of a set of neurons 
+:math:`\{x_i | x_1, x_2, ..., x_m\}` representing the input features. Each hidden
+layer transforms the values from the previous layer by a weighted linear summation
+:math:`w_1x_1 + w_2x_2 + ... + w_mx_m`, followed by a non-linear activation function
+:math:`g(\cdot):R \rightarrow R` - like the hyperbolic tan function. The output layer
+receives the values from the last hidden layer and transforms them into output values.
+
+The module contains the public attributes ``coefs_`` and ``intercepts_``.
+``coefs_`` is a list of weight matrices, where weight matrix at index
+:math:`i` represents the weights between layer :math:`i` and layer 
+:math:`i+1`. ``intercepts_`` is a list of bias vectors, where the vector
+at index :math:`i` represents the bias values added to layer :math:`i+1`.
+
+The advantages of Multi-layer Perceptron are:
+
+    + Capability to learn complex/non-linear models.
+
+    + Capability to learn models in real-time (on-line learning) 
+      using ``partial_fit``.
+      
+      
+The disadvantages of Multi-layer Perceptron (MLP) include:
+
+    + MLP with hidden layers have a non-convex loss function where there exists 
+      more than one local minimum. Therefore different random weight 
+      initializations can lead to different validation accuracy.
+
+    + MLP suffers from the Backpropagation diffusion problem; layers far from 
```

Comment:
@amueller I think I have time for this pull request. 

I'm really sorry that I just browsed the documentation after I took over, and it's probably due to some misunderstanding on my side.
I thought the documentation must have been reviewed several times, since the previous two pull requests had been around for quite some time when I took over. Because of that impression in my mind, I kind of assumed that everything else is mostly correct except for the optimization part I was supposed to modify.

I'll carefully read everything again.

BTW, some of the checks seemed to be failing before even actually starting. Is this normal?